### ISSUE TITLE
Create in-app Terms of Use

### ISSUE THREAD

**Issue by kevinghim** [MetaMask/metamask-mobile#5009]
### Background
As a legal requirement, we will require MetaMask users to agreed to the Terms of Use before using the app. That is, get explicit user acknowledgment of our Terms of Use, by viewing the entire Terms of Use, clicking to agree to the Terms of Use to accept and use the app.

### Designs
[Final designs](https://www.figma.com/file/j3T1H0Xiuke9NhFJ3yOxBU/Updated-Terms-of-Use?node-id=87%3A5216)

https://user-images.githubusercontent.com/6951929/191128485-9e703139-2b9c-414c-b08c-09beaa9376f4.mov

### Acceptance Criteria
Use the Terms of Use using the "What's New" modal
Display the Terms of Use to all existing users
For other users, display the Terms of Use before someone creates a wallet or imports a wallet (that is, before someone uses a seedphrase in MM)
Display once per wallet

**Comment by sethkfman** [comment]
Use the Terms of Use using the "What's New" modal.

**Comment by sethkfman** [comment]
@kevinghim We do need more explanation:

**Comment by kevinghim** [comment]
@sethkfman  answers in-line

> * Where is the copy?
[Terms of Use](https://consensys.net/terms-of-use/)

> * Does this need to be translated?
Yes

> * In this statement. Do you mean we need to add a what's new modal that requires a user to acknowledge the "Terms of Use"
Does mobile use the "What's New" modal? My thought here was to utilize that modal to display the ToU.

**Comment by omnat** [comment]
@kevinghim Is the attached recording what you intended to add? It looks more related to transaction experience.

**Comment by kevinghim** [comment]
> @kevinghim Is the attached recording what you intended to add? It looks more related to transaction experience.

Sorry that was the wrong clip. I've updated it.

**Comment by sethkfman** [comment]
@kevinghim We do have a "What's new" modal. I see what you are saying by using that modal. Thanks for the update to the video

**Comment by sethkfman** [comment]
@kevinghim

**Comment by kevinghim** [comment]
> Can the Terms of Use be referenced from the website or another dynamic source? We don't want to have to compile it into the app.
Yes as long as we have guarantees it'll load the same ToU consistently

> If we are requiring translations, has the legal team provided those translations? I would not want our translation service to do that.
I'll find out with the legal team.

> Can users by-pass accepting the Terms of Use (e.g. should it be a blocker)?
It should be a blocker. To use the app, they have to agree to terms.

**Comment by sethkfman** [comment]
@kevinghim Do you want us to reference the ConsenSys page for agreeing to the Terms of Use? Similar, how we do it now on mobilehttps://images.zenhubusercontent.com/141427485/1c4ad1ce-b81a-454b-be7d-5823f2b6ff1b/simulator_screen_recording___iphone_13___2022_09_26_at_16_23_51.mp4?

**Comment by kevinghim** [comment]
@sethkfman extension is going to have the entire text within the modal instead of referencing it. This is the preference from Legal, and we should aim for consistency. We should also make sure that the modal is not closed by clicking outside of it, and that the only way to use the app is through accepting the terms.

**Comment by sethkfman** [comment]
[ToU-en.txt](https://github.com/MetaMask/metamask-mobile/files/9711220/ToU-en.txt)

**Comment by sethkfman** [comment]
@kevinghim @adonesky1 I have a PR started to fetch the translations for the Terms Of Use. I don't know if Blend should translating legal documents but I have started the process.
